perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -1739,20 +1740,27 @@ protected boolean hasCandidateExistingMember(Candidate candidate, RichMember ric
 		return skippedGroups;
 	}
 
-	/**
-	 * Adds the group on the first place to the queue of groups waiting for synchronization.
-	 *
-	 * @param group the group to be forced this way
-	 *
-	 * @throws GroupSynchronizationAlreadyRunningException when group synchronization is already running at this moment
-	 */
 	@Override
-	public void forceGroupSynchronization(PerunSession sess, Group group) throws GroupSynchronizationAlreadyRunningException {
-		//Check if the group is not currently in synchronization process
-		if(poolOfSynchronizations.putGroupToPoolOfWaitingGroups(group, true)) {
-			log.debug("Scheduling synchronization for the group {} by force!", group);
+	public void forceGroupSynchronization(PerunSession sess, Group group) throws GroupSynchronizationAlreadyRunningException, GroupSynchronizationNotEnabledException {
+		//Check if the group should be synchronized (attribute synchronizationEnabled is set to 'true')
+		Boolean syncEnabled = false;
+		try {
+			Attribute syncEnabledAttr = getPerunBl().getAttributesManagerBl().getAttribute(sess, group, GroupsManager.GROUPSYNCHROENABLED_ATTRNAME);
+			if(syncEnabledAttr.getValue() != null) syncEnabled = syncEnabledAttr.valueAsBoolean();
```

Comment:
This attribute is not boolean but String so I do not think you can cast it to boolean like that.

Reply:
You are right, fixed.